tests/test_minimizers.py

```diff
@@ -73,346 +80,334 @@ def subclasses(base, leaves_only=True):
 
-class TestMinimize(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls):
-        np.random.seed(0)
 
-    def test_custom_objective(self):
-        """
-        Compare the result of a custom objective with the symbolic result.
-        :return:
-        """
-        # Create test data
-        xdata = np.linspace(0, 100, 25)  # From 0 to 100 in 100 steps
-        a_vec = np.random.normal(15.0, scale=2.0, size=xdata.shape)
-        b_vec = np.random.normal(100, scale=2.0, size=xdata.shape)
-        ydata = a_vec * xdata + b_vec  # Point scattered around the line 5 * x + 105
-
-        # Normal symbolic fit
-        a = Parameter('a', value=0, min=0.0, max=1000)
-        b = Parameter('b', value=0, min=0.0, max=1000)
-        x = Variable('x')
-        y = Variable('y')
-        model = a * x + b
-
-        fit = Fit(model, xdata, ydata, minimizer=BFGS)
-        fit_result = fit.execute()
-
-        def f(x, a, b):
-            return a * x + b
-
-        def chi_squared(a, b):
-            return np.sum((ydata - f(xdata, a, b))**2)
-
-        with warnings.catch_warnings(record=True) as w:
-            # Should no longer raise warnings, because internally we practice
-            # what we preach.
-            warnings.simplefilter("always")
-            fit_custom = BFGS(chi_squared, [a, b])
-            self.assertTrue(len(w) == 0)
-
-        fit_custom_result = fit_custom.execute()
-
-        self.assertIsInstance(fit_custom_result, FitResults)
-        self.assertAlmostEqual(fit_custom_result.value(a) / fit_result.value(a), 1.0, 5)
-        self.assertAlmostEqual(fit_custom_result.value(b) / fit_result.value(b), 1.0, 4)
-
-        # New preferred usage, multi component friendly.
-        with self.assertRaises(TypeError):
-            callable_model = CallableNumericalModel(
-                chi_squared,
-                connectivity_mapping={y: {a, b}}
-            )
+def setup_function():
+    np.random.seed(0)
+
+
+def test_custom_objective():
+    """
+    Compare the result of a custom objective with the symbolic result.
+    :return:
+    """
+    # Create test data
+    xdata = np.linspace(0, 100, 25)  # From 0 to 100 in 100 steps
+    a_vec = np.random.normal(15.0, scale=2.0, size=xdata.shape)
+    b_vec = np.random.normal(100, scale=2.0, size=xdata.shape)
+    ydata = a_vec * xdata + b_vec  # Point scattered around the line 5 * x + 105
+
+    # Normal symbolic fit
+    a = Parameter('a', value=0, min=0.0, max=1000)
+    b = Parameter('b', value=0, min=0.0, max=1000)
+    x = Variable('x')
+    y = Variable('y')
+    model = a * x + b
+
+    fit = Fit(model, xdata, ydata, minimizer=BFGS)
+    fit_result = fit.execute()
+
+    def f(x, a, b):
+        return a * x + b
+
+    def chi_squared(a, b):
+        return np.sum((ydata - f(xdata, a, b))**2)
+
+    with warnings.catch_warnings(record=True) as w:
+        # Should no longer raise warnings, because internally we practice
+        # what we preach.
```

Comment:
I don't understand why/what this does. Maybe @tBuLi remembers? But not for this PR.

Reply:
Not sure, but I think it is because we changed the API for directly using the minimizers to raise a deprecation warning when used old school. This is just to make sure our test is correct. Should of course be changed to a test for the deprecation warning instead...

Reply:
So how can I handle this one? :D

Reply:
Ignore it for now.